src/cpu/aarch64/jit_uni_reorder.cpp

```diff
@@ -163,12 +163,15 @@ struct jit_uni_reorder_kernel_f32_t : public kernel_t, public jit_generator {
 
         bool ok = true && p.ndims > 0
                 && utils::one_of(p.itype, f32, s32, data_type::s8, u8)
-                && utils::one_of(p.otype, f32, s32, data_type::s8, u8)
+                && utils::one_of(p.otype, f32, bf16, s32, data_type::s8, u8)
                 && utils::everyone_is(0, p.ioff, p.ooff) /* do we need this? */
                 && utils::one_of(p.beta, 0.f, 1.f) /* anything else? */
                 && simple_impl_desc_init(p, nullptr)
                 && prb_has_small_strides(p);
 
+        if (ok && p.otype == bf16)
+            ok = ok && p.itype == f32 && mayiuse(feat_bf16);
```

Comment:
```suggestion
            ok = ok && p.itype == f32 && mayiuse_bf16() && mayiuse(sve_128);
```
Cases that support BF16 but do not support SVE must be excluded. (Apple M2 doesn't supports SVE, but BF16?)

Reply:
Hi @kawakami-k, it is sufficient only to check for BF16 because there are BF16 instructions that are available in NEON and SVE and with change that you proposed to use `bfcvtn` should not depend solely on SVE anymore.

Reply:
Would this step be a problem, if SVE is not checked?
https://github.com/oneapi-src/oneDNN/pull/1594/files/faf773e6d2df7cc76e04c91f7f18718280ef7ab3#diff-cc4e1c6ea12d36d422f4cb9ffc4d0ed584856d1e10913c674a445f7fed93a793R1293

Reply:
> Would this step be a problem, if SVE is not checked?
> https://github.com/oneapi-src/oneDNN/pull/1594/files/faf773e6d2df7cc76e04c91f7f18718280ef7ab3#diff-cc4e1c6ea12d36d422f4cb9ffc4d0ed584856d1e10913c674a445f7fed93a793R1293

Hi @kawakami-k, thanks for spotting this. I have removed in the most recent commit use of `st1h` instruction as we do not need it and can simply use `store` method since we are using `bfcvtn` so when doing downcast from FP32 to BF16 we are not using any of SVE instructions.